tests/components/telegram_bot/conftest.py

```diff
@@ -0,0 +1,190 @@
+                "first_name": "Firstname",
+                "username": "some_username",
+                "language_code": "en",
+            },
+            "chat": {
+                "id": -123456789,
+                "title": "SomeChat",
+                "type": "group",
+                "all_members_are_administrators": True,
+            },
+            "date": 1644518189,
+            "text": "/command",
+            "entities": [
+                {
+                    "type": "bot_command",
+                    "offset": 0,
+                    "length": 7,
+                }
+            ],
+        },
+    }
+
+
+@pytest.fixture
+def update_message_text():
+    """Fixture for mocking an incoming update of type message/text."""
+    return {
+        "update_id": 1,
+        "message": {
+            "message_id": 1,
+            "date": 1441645532,
+            "from": {
+                "id": 12345678,
+                "is_bot": False,
+                "last_name": "Test Lastname",
+                "first_name": "Test Firstname",
+                "username": "Testusername",
+            },
+            "chat": {
+                "last_name": "Test Lastname",
+                "id": 1111111,
+                "type": "private",
+                "first_name": "Test Firstname",
+                "username": "Testusername",
+            },
+            "text": "HELLO",
+        },
+    }
+
+
+@pytest.fixture
+def update_callback_query():
+    """Fixture for mocking an incoming update of type callback_query."""
+    return {
+        "update_id": 1,
+        "callback_query": {
+            "id": "4382bfdwdsb323b2d9",
+            "from": {
+                "id": 12345678,
+                "type": "private",
+                "is_bot": False,
+                "last_name": "Test Lastname",
+                "first_name": "Test Firstname",
+                "username": "Testusername",
+            },
+            "chat_instance": "aaa111",
+            "data": "Data from button callback",
+            "inline_message_id": "1234csdbsk4839",
+        },
+    }
+
+
+@pytest.fixture
+async def webhook_platform(hass, config_webhooks, mock_register_webhook):
+    """Fixture for setting up the webhooks platform using appropriate config and mocks."""
+    await async_setup_component(
+        hass,
+        DOMAIN,
+        config_webhooks,
+    )
+    await hass.async_block_till_done()
+
+
+@pytest.fixture
+async def polling_platform(hass, config_polling):
+    """Fixture for setting up the polling platform using appropriate config and mocks."""
+    await async_setup_component(
+        hass,
+        DOMAIN,
+        config_polling,
+    )
+    await hass.async_block_till_done()
+
+
+@pytest.fixture
+def pushbot(hass, config_webhooks):
+    """Fixture for an initialized `PushBot`."""
+    platform_config = config_webhooks[DOMAIN][0]
+    bot = initialize_bot(platform_config)
+    pushbot = PushBot(hass, bot, platform_config)
```

Comment:
We don't want to access integration details in the tests. Try to patch the library as needed instead, eg to be able to trigger callbacks.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
Fair enough, but I have to say it seems less robust to base tests on library integration details.